### PR TITLE
local imports

### PR DESCRIPTION
**Pull request by IAD**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -59,17 +60,26 @@ func main() {
...
+		f, err = open(src, cli)
+		if err != nil {
+			return err
+		}
+		filePath = filepath.Dir(src)
+		if err != nil {
+			return err
+		}
 	}
+	log.Println("filePath", filePath)
```
Comment (fiorix) [inline main.go]: this doesn't belong here :)

File: main.go [hunk 2]
```diff
@@ -59,17 +60,26 @@ func main() {
...
 	d, err := wsdl.Unmarshal(f)
 	if err != nil {
 		return err
 	}
 	f.Close()
-	enc := wsdlgo.NewEncoder(w)
+	enc := wsdlgo.NewEncoder(w, filePath)
```
Comment (fiorix) [inline main.go]: how about using [`os.Chdir`](https://godoc.org/os#Chdir) here instead? then the import function opens files relative to the main file

File: wsdlgo/encoder.go [hunk 3]
```diff
@@ -40,6 +41,9 @@ type goEncoder struct {
 	// where to write Go code
 	w io.Writer
 
+	// root file absolute path
+	rootFileAbsolutePath string
```
Comment (fiorix) [inline wsdlgo/encoder.go]: no need for this if we use chdir ^^

File: wsdlgo/encoder.go [hunk 4]
```diff
@@ -244,8 +249,18 @@ func (ge *goEncoder) importSchema(d *wsdl.Definitions) error {
 }
 
 // download xml from url, decode in v.
-func (ge *goEncoder) importRemote(url string, v interface{}) error {
-	resp, err := ge.http.Get(url)
+func (ge *goEncoder) importRemote(name string, v interface{}) error {
+	u, err := url.Parse(name)
+	if err != nil || u.Scheme == "" {
+		file, err := os.Open(filepath.Join(ge.rootFileAbsolutePath, name))
```
Comment (fiorix) [inline wsdlgo/encoder.go]: no need for filepath.Join if we use chdir ^^

File: wsdlgo/encoder_test.go [hunk 5]
```diff
@@ -17,7 +17,7 @@ import (
 )
 
 func LoadDefinition(t *testing.T, filename string, want error) *wsdl.Definitions {
-	f, err := os.Open(filepath.Join("testdata", filename))
+	f, err := os.Open(GetDefinitionFilePath(filename))
```
Comment (fiorix) [inline wsdlgo/encoder_test.go]: what's wrong with just `filepath.Join` here? the other function only calls that anyway

**Comment by IAD** [comment]
agree

**Comment by IAD** [comment]
Hello @fiorix I propose to leave previous version.
Solution with os.Chdir(is a side effect) could work only with simple includes. It will stop working with recursive includes.

if we have something like:
/root.wsdl
/dir1/file.xml
/dir1/dir2/file2.xml

- root.wsdl includes file1.xml
- file1.xml includes file2.xml

**Comment by IAD** [comment]
@fiorix updated, please review. There are a solution compatible with current constructor contract

File: main.go (hunk 2 above)
Comment (IAD) [inline main.go]: proposed soltion without side effects

**Comment by fiorix** [comment]
You should only call `os.Chdir` once, to the directory where the first WSDL file is located. Then, you don't have to do anything else, assuming that included files are relative to the file being processed.

As a security measure we should not support including files with absolute paths, that is, failing if the include path starts with `/`.

**Pull request closed without merging**